packages/compass-aggregations/src/modules/stage.spec.js

```diff
@@ -310,20 +431,21 @@ describe('Stage module', () => {
       it('generates timestamp', () => {
         expect(generated.ts.low).to.equal(-321469502);
       });
+
       it('returns the stage string', () => {
         expect(generateStageAsString(stage)).to.deep.equal(`{$match: {
-  code: Code('some code'),
-  oid: ObjectId('5a7382114ec1f67ae445f778'),
-  bin: Binary('aakjadfjadfldksjfadf', '1'),
-  dbref: DBRef('db.coll', '1'),
-  nl: NumberLong('3'),
-  nd: NumberDecimal('5.00000001'),
-  ni: NumberInt('5'),
-  minkey: MinKey(),
-  maxkey: MaxKey(),
-  isodate: ISODate('1999-01-01'),
-  regexp: RegExp('/^[a-z0-9_-]{3,16}$/'),
-  ts: Timestamp(-321469502, 367)
+ code: Code('some code'),
+ oid: ObjectId('5a7382114ec1f67ae445f778'),
+ bin: BinData(1, 'YWFramFkZmphZGZsZGtzamZhZGY='),
```

Comment:
hmmm

Reply:
Should be fine? https://docs.mongodb.com/manual/reference/mongodb-extended-json-v1/#mongodb-bsontype-data_binary

Reply:
It’s actually a lot better, because the JS BSON `Binary()` constructor takes a ISO-8859-1 string as input, which will usually look like garbage when it’s used to represent generic binary data. Base64 makes that a lot nicer :+1:

Reply:
That being said, we did make the intentional decision in mongosh to move people towards `Binary` :woman_shrugging:

Reply:
The only thing that gives me pause is that the test's input uses Binary https://github.com/mongodb-js/compass/blob/fe722a299da2f02613485a34dffcdee308585fc1/packages/compass-aggregations/src/modules/stage.spec.js#L366

I'm not sure what where is transforming that, though.

Reply:
aah from here: https://github.com/mongodb-js/query-parser/blob/406ed4ec774b9dea707c39b694e67d8d085545bc/lib/stringify.js#L49-L55